contracts/test/unit/VoltV2.t.sol

```diff
@@ -0,0 +1,384 @@
+
+    Vm private vm = Vm(HEVM_ADDRESS);
+
+    function setUp() public {
+        core = getCore();
+        volt = new VoltV2(address(core));
+    }
+
+    function testTokenDetails() public {
+        assertEq(volt.name(), "Volt");
+        assertEq(volt.symbol(), "VOLT");
+        assertEq(volt.decimals(), 18);
+        assertEq(volt.totalSupply(), 0);
+    }
+
+    function testDelegate() public {
+        volt.delegate(address(0xFFF));
+        assertEq(volt.delegates(address(this)), address(0xFFF));
+    }
+
+    function testMintSuccessMinter(uint224 voltToMint) public {
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(0xFFF), voltToMint);
+
+        vm.prank(address(0xFFF));
+        volt.delegate(address(0xFFF));
+
+        assertEq(volt.totalSupply(), voltToMint);
+        assertEq(volt.getCurrentVotes(address(0xFFF)), voltToMint);
+        assertEq(volt.balanceOf(address(0xFFF)), voltToMint);
+    }
+
+    function testMintAfterDelegation(uint224 voltToMint) public {
+        vm.assume(voltToMint < type(uint224).max / 2);
+
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(0xFFF), voltToMint);
+
+        vm.prank(address(0xFFF));
+        volt.delegate(address(0xFFF));
+
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(0xFFF), voltToMint);
+
+        assertEq(volt.getCurrentVotes(address(0xFFF)), voltToMint * 2);
+    }
+
+    function testMintFailureUnauthorized() public {
+        vm.startPrank(address(0xFFF));
+        vm.expectRevert("CoreRef: Caller is not a minter");
+        volt.mint(address(0xFFF), 1e18);
+        vm.stopPrank();
+    }
+
+    function testMintFailToVoltContract() public {
+        vm.startPrank(addresses.minterAddress);
+        vm.expectRevert("Volt: cannot transfer to the volt contract");
+        volt.mint(address(volt), 1e18);
+        vm.stopPrank();
+    }
+
+    function testMintFailZeroAddress() public {
+        vm.startPrank(addresses.minterAddress);
+        vm.expectRevert("Volt: cannot transfer to the zero address");
+        volt.mint(address(0), 1e18);
+        vm.stopPrank();
+    }
+
+    function testBurn(uint224 voltToBurn) public {
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(this), voltToBurn);
+        volt.delegate(address(this));
+        assertEq(volt.getCurrentVotes(address(this)), voltToBurn);
+
+        volt.burn(voltToBurn);
+
+        assertEq(volt.totalSupply(), 0);
+        assertEq(volt.balanceOf(address(this)), 0);
+        assertEq(volt.getCurrentVotes(address(this)), 0);
+    }
+
+    function testBurnFail() public {
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(this), 1e18);
+
+        vm.expectRevert("Volt: burn amount exceeds balance");
+        volt.burn(2e18);
+    }
+
+    function testBurnFrom(uint224 voltToBurn) public {
+        vm.assume(voltToBurn < type(uint224).max); // to make sure we don't run into infinite approval counter example
```

Comment:
It's fine to do infinite approvals, and that would be good to test in this fuzz test.

Instead, when we assert allowance, if voltToBurn is uint224 max, then we should assert that allowance stays the same.